### PR TITLE
fix: Use SEMAPHORE_GIT_WORKING_BRANCH to associate Cloud recording to current branch

### LINKED ISSUE #26309: Cypress Cloud Uses Wrong Branch in Semaphore CI
### Current behavior

In Sempahore CI, on pull requests, Cypress Cloud test runs show up under the target branch.

### Desired behavior

Test runs should show up as being run on the current branch.

### Test code to reproduce

Cypress uses `SEMAPHORE_GIT_BRANCH` as the branch to report test runs.  However, for a pull request this points to target branch.  This is a problem, because we want the current branch.

https://github.com/cypress-io/cypress/blob/a5b0cad9f0b7a49674c79747cc2616e788a3b2da/packages/server/lib/util/ci_provider.js#L573

https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_branch

### Cypress Version

12.8.1

### Node version

16.13.0

### Operating System

Ubuntu 20.04.6 LTS (Focal Fossa)

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by leightjohnson93**
- Closes #26309 

In Sempahore CI, on pull requests, Cypress Cloud test runs show up under the target branch, when run should be associated to the current branch.

Fix Cypress to use the correct Semphore env to report the current branch. See https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_branch

### PR Tasks

- [n/a] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ _Released 04/11/2023 (PENDING)_
 
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
 
+ - Use correct branch for test results in Semaphore CI. Fixed in [#26333](https://github.com/cypress-io/cypress/pull/26333).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
 - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/pull/26309).
```

**Comment by emilyrohrbough** [comment]
@leightjohnson93 Thank you so much for this fix! We appreciate this! I updated your PR title & the summary for others on the team and linked your issue to close it out. Overall looks good. 👍🏻

**Review by emilyrohrbough — APPROVED** [review]

**Review by nagash77 — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ _Released 04/11/2023 (PENDING)_
 
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
 
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/pull/26309).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
 - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
```

**Comment by emilyrohrbough** [comment]
@leightjohnson93 Could you update the [`ci_provider_spec.js`](https://github.com/cypress-io/cypress/blob/develop/packages/server/test/unit/util/ci_provider_spec.js) unit test?

**Comment by leightjohnson93** [comment]
No problem, can do that Tuesday

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -8,14 +8,16 @@ _Released 04/11/2023 (PENDING)_
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
  - Fixed an issue in the onboarding wizard where project framework & bundler would not be auto-detected when opening directly into component testing mode using the `--component` CLI flag. Fixes [#22777](https://github.com/cypress-io/cypress/issues/22777).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
```

File: cli/CHANGELOG.md [hunk 4]
```diff
@@ -8,14 +8,16 @@ _Released 04/11/2023 (PENDING)_
...
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
 
 **Dependency Updates:**
 
 - Upgraded [`minimist`](https://www.npmjs.com/package/minimist) from `1.2.6` to `1.2.8` to address this [CVE-2021-44906](https://github.com/advisories/GHSA-xvch-5gv4-984h) NVD security vulnerability. Addressed in [#26254](https://github.com/cypress-io/cypress/pull/26254).
 
+**Misc:**
+
+- Removed unintentional debug logs. Address in [#26411](https://github.com/cypress-io/cypress/pull/26411)
+
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
```

File: cli/CHANGELOG.md [hunk 5]
```diff
@@ -8,14 +8,16 @@ _Released 04/11/2023 (PENDING)_
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
  - Fixed an issue in the onboarding wizard where project framework & bundler would not be auto-detected when opening directly into component testing mode using the `--component` CLI flag. Fixes [#22777](https://github.com/cypress-io/cypress/issues/22777).
 
-**Misc:**
-
-- Removed unintentional debug logs. Addressed in [#26411](https://github.com/cypress-io/cypress/pull/26411).
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md:10]:
```suggestion
 - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).

 **Misc:**

- Removed unintentional debug logs. Address in [#26411](https://github.com/cypress-io/cypress/pull/26411)

```

File: cli/CHANGELOG.md [hunk 6]
```diff
@@ -7,10 +7,12 @@ _Released 04/11/2023 (PENDING)_
 
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
  - Fixed an issue in the onboarding wizard where project framework & bundler would not be auto-detected when opening directly into component testing mode using the `--component` CLI flag. Fixes [#22777](https://github.com/cypress-io/cypress/issues/22777).
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md:10]:
```suggestion
 - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).

```

File: cli/CHANGELOG.md [hunk 7]
```diff
@@ -7,10 +7,13 @@ _Released 04/11/2023 (PENDING)_
...
 
-**Misc:**
 
-- Removed unintentional debug logs. Addressed in [#26411](https://github.com/cypress-io/cypress/pull/26411).
+ **Misc:**
+
+- Removed unintentional debug logs. Address in [#26411](https://github.com/cypress-io/cypress/pull/26411)
+
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
**Misc:**

- Removed unintentional debug logs. Address in [#26411](https://github.com/cypress-io/cypress/pull/26411)
```

File: cli/CHANGELOG.md [hunk 8]
```diff
@@ -7,10 +7,12 @@ _Released 04/11/2023 (PENDING)_
 
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
  - Fixed an issue in the onboarding wizard where project framework & bundler would not be auto-detected when opening directly into component testing mode using the `--component` CLI flag. Fixes [#22777](https://github.com/cypress-io/cypress/issues/22777).
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
+
 
 **Misc:**
 
-- Removed unintentional debug logs. Addressed in [#26411](https://github.com/cypress-io/cypress/pull/26411).
+- Removed unintentional debug logs. Address in [#26411](https://github.com/cypress-io/cypress/pull/26411)
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
- Removed unintentional debug logs. Addressed in [#26411](https://github.com/cypress-io/cypress/pull/26411).
```

File: cli/CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,8 @@ _Released 04/11/2023 (PENDING)_
 
  - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
  - Fixed an issue in the onboarding wizard where project framework & bundler would not be auto-detected when opening directly into component testing mode using the `--component` CLI flag. Fixes [#22777](https://github.com/cypress-io/cypress/issues/22777).
+ - Updated to use the `SEMAPHORE_GIT_WORKING_BRANCH` [Semphore](https://docs.semaphoreci.com) CI environment variable to correctly associate a Cloud run to the current branch. Previously this was incorrectly associating a run to the target branch. Fixes [#26309](https://github.com/cypress-io/cypress/issues/26309).
+
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
```

**Pull request merged**